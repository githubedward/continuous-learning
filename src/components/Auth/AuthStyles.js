import styled from "styled-components";
import { appear } from "../shared/style-variables";

// styles
export const Form = styled.form`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 3.5rem 3rem;
  background: #ffffff95;
  border-radius: 3px;
  animation: ${appear} 500ms ease-in;
`;
export const Title = styled.h1`
  margin-bottom: 1rem;
`;
export const Label = styled.label`
  margin: 0.5rem;
`;
export const Input = styled.input`
  padding: 0.2rem;
  font-size: 1.25rem;
`;
export const Button = styled.button`
  margin-top: 3rem;
  color: #ffffff;
  font-size: 1rem;
  width: 7.5rem;
  padding: 0.5rem 1rem;
  background: #fb720e;
  border: 1px solid #fb720e;
  border-radius: 3px;
  text-decoration: none;
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  &:hover {
    background: #fd8912;
    transition: 0.3s;
  }
`;
export const Span = styled.span`
  color: #fd8912;
`;
export const Error = styled.span`
  color: red;
  margin-top: 0.5rem;
  font-weight: 300;
`;
