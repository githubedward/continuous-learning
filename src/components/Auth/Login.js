import React from "react";
import PropTypes from "prop-types";
import {
  Container,
  Title,
  Form,
  Label,
  Input,
  Button,
  Span
} from "./AuthStyles";

const Login = ({ handleUsername, handlePassword, username, password }) => {
  return (
    <Container>
      <Form>
        <Title>
          <Span>Login</Span>
        </Title>
        <Label>
          Username:
          <Input
            type="text"
            name="username"
            value={username}
            onChange={handleUsername}
          />
        </Label>
        <Label>
          Password:
          <Input
            type="password"
            name="password"
            value={password}
            onChange={handlePassword}
          />
        </Label>
        <Button>Login</Button>
      </Form>
    </Container>
  );
};

Login.propTypes = {
  handleUsername: PropTypes.func.isRequired,
  handlePassword: PropTypes.func.isRequired,
  username: PropTypes.string.isRequired,
  password: PropTypes.string.isRequired
};

Login.defaultProps = {
  handleUsername: () => alert("default"),
  handlePassword: () => alert("default"),
  username: "default",
  password: "default"
};

export default Login;
