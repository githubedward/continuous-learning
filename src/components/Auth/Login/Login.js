import React from "react";
import PropTypes from "prop-types";
import { H1, Form, Input, Span, Footer, Status } from "../AuthStyles";
import Anchor from "../../shared/Anchor";
import PageContainer from "../../shared/PageContainer";
import BrandName from "../../shared/BrandName";
import * as styleGuides from "../../shared/style-variables";
import { MagicSpinner } from "react-spinners-kit";
import Button from "../../shared/Button";
import { ThemeProvider } from "styled-components";
import { btnTheme } from "../AuthStyles";

const Login = ({
  handleUsername,
  handlePassword,
  handleLogin,
  linkSignup,
  isLoading,
  username,
  password,
  status
}) => {
  const data = {
    username,
    password
  };

  return (
    <PageContainer>
      <BrandName size={"3.5rem"} />
      <H1>
        <Span color={styleGuides.gray}>
          <Span fontWeight="bold" color={styleGuides.gray}>
            Login
          </Span>{" "}
          to find and share beautiful experiences
        </Span>
      </H1>
      <Form
        onSubmit={e => {
          e.preventDefault();
          handleLogin(data);
        }}
      >
        <Input
          type="text"
          name="username"
          placeholder="Username"
          value={username}
          onChange={e => {
            handleUsername(e.target.value);
          }}
        />
        <Input
          type="password"
          name="password"
          placeholder="Password"
          value={password}
          onChange={e => {
            handlePassword(e.target.value);
          }}
        />
        <Status>{status}</Status>
        <ThemeProvider theme={btnTheme}>
          <Button>
            {isLoading ? (
              <MagicSpinner loading={isLoading} size={25} color={"white"} />
            ) : (
              "Login"
            )}
          </Button>
        </ThemeProvider>
      </Form>
      <Footer>
        <Span color={styleGuides.darkgray} fontWeight={styleGuides.lighter}>
          Don't have an account?
        </Span>{" "}
        <Anchor onClick={linkSignup}>Signup</Anchor>
      </Footer>
    </PageContainer>
  );
};

Login.propTypes = {
  handleUsername: PropTypes.func.isRequired,
  handlePassword: PropTypes.func.isRequired,
  linkSignup: PropTypes.func.isRequired,
  username: PropTypes.string.isRequired,
  password: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired
};

Login.defaultProps = {
  handleUsername: () => alert("default"),
  handlePassword: () => alert("default"),
  linkSignup: () => alert("default"),
  username: "default",
  password: "default",
  status: "default"
};

export default Login;
