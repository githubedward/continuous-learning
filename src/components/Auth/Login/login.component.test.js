import Login from "./Login";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";
import React from "react";
import Enzyme, { shallow, mount } from "enzyme";
import { MagicSpinner } from "react-spinners-kit";
import { Status } from "../AuthStyles";

Enzyme.configure({ adapter: new Adapter() });

describe("<Login />", () => {
  const props = {
    handleUsername: jest.fn(),
    handlePassword: jest.fn(),
    handleLogin: jest.fn(),
    linkSignup: jest.fn(),
    isLoading: false,
    username: "username",
    password: "password",
    status: "status"
  };

  // shallow rendering
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Login {...props} />);
  });

  it("should render without errors", () => {
    expect(wrapper).toBeTruthy();
  });

  it("should match the snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  // Full DOM rendering
  let component;
  beforeEach(() => {
    component = mount(<Login {...props} />);
  });

  it("should render button correctly", () => {
    const button = component.find("button");
    expect(button.text()).toBe("Login");
    props.isLoading = true;
    expect(button.find(MagicSpinner)).toBeTruthy();
  });

  it("should handle username input change", () => {
    const input = component
      .find("input")
      .filterWhere(el => el.prop("name") === "username");
    expect(input.prop("placeholder")).toBe("Username");
    expect(input.prop("type")).toBe("text");
    input.simulate("change");
    expect(component.props().handleUsername).toHaveBeenCalled();
  });

  it("should handle password input change", () => {
    const input = component
      .find("input")
      .filterWhere(el => el.prop("name") === "password");
    expect(input.prop("placeholder")).toBe("Password");
    expect(input.prop("type")).toBe("password");
    input.simulate("change");
    expect(component.props().handlePassword).toHaveBeenCalled();
  });

  it("should call fn when form is submitted", () => {
    const form = component.find("form");
    form.simulate("submit");
    expect(props.handleLogin).toHaveBeenCalled();
  });

  it("should render link", () => {
    expect(component.find("a")).toBeTruthy();
  });

  it("should update status", () => {
    const status = component.find(Status);
    expect(status.text()).toBe("status");
    component.setProps({ status: "test" });
    expect(status.text()).toBe("test");
  });
});
