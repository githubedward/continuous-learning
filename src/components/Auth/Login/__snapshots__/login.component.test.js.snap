// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should match the snapshot 1`] = `
<styled.section>
  <BrandName
    size="3.5rem"
  />
  <styled.h1>
    <styled.span
      color="#B4B4B4"
    >
      <styled.span
        color="#B4B4B4"
        fontWeight="bold"
      >
        Login
      </styled.span>
       
      to find and share beautiful experiences
    </styled.span>
  </styled.h1>
  <styled.form
    onSubmit={[Function]}
    padding="2rem 2.5rem"
  >
    <styled.input
      appearance="none"
      border="1px solid #B4B4B4"
      name="username"
      onChange={[Function]}
      placeholder="Username"
      type="text"
      value="username"
    />
    <styled.input
      appearance="none"
      border="1px solid #B4B4B4"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value="password"
    />
    <styled.span>
      status
    </styled.span>
    <ThemeProvider
      theme={
        Object {
          "bgc": "#F3615C",
          "borderColor": "#F3615C",
          "color": "white",
          "fontSize": "1.5rem",
          "fweight": 500,
          "height": "3rem",
          "hoverbgc": "#ed4949",
          "margin": "3rem 0 0 0",
          "padding": ".5rem 1rem",
          "width": "inherit",
        }
      }
    >
      <styled.button>
        Login
      </styled.button>
    </ThemeProvider>
  </styled.form>
  <styled.footer>
    <styled.span
      color="#888888"
      fontWeight={100}
    >
      Don't have an account?
    </styled.span>
     
    <styled.a
      onClick={[MockFunction]}
    >
      Signup
    </styled.a>
  </styled.footer>
</styled.section>
`;
